tests/test_js_functions.py

```diff
@@ -57,6 +57,38 @@ def test_createElement():
     document.createElement();
     """)
 
+def test_enablePrivilege():
+    """Tests that enablePrivilege throws a warning."""
+
+    err = _do_test_raw("""
+        netscape.security.PrivilegeManager.enablePrivilege()
+    """)
+    assert err.warnings
+
+    eq_(err.warnings[0]["id"],
+        ("js", "traverser", "dangerous_global"))
+    eq_(err.warnings[0].get("signing_severity"), "high")
```

Comment:
On Tue, May 05, 2015 at 07:07:48AM -0700, Mathieu Agopian wrote:
>> @@ -57,6 +57,38 @@ def test_createElement():
>>      document.createElement();
>>      """)
>>
>> +def test_enablePrivilege():
>> +    """Tests that enablePrivilege throws a warning."""
>> +
>> +    err = _do_test_raw("""
>> +        netscape.security.PrivilegeManager.enablePrivilege()
>> +    """)
>> +    assert err.warnings
>> +
>> +    eq_(err.warnings[0]["id"],
>> +        ("js", "traverser", "dangerous_global"))
>> +    eq_(err.warnings[0].get("signing_severity"), "high")
>
>why a `.get()` here?

Not all messages have a `"signing_severity"` property. I guess 
the `KeyError` would make the test fail the same as the 
`AssertionError`, I just tend to avoid letting tests fail that 
way.